### PR TITLE
Add support for Node.js v20

### PR DESCRIPTION
**Pull request by afharo**
Running the tests in Node.js v20 slightly changes the traces so I created a `v20` directory inside `expected` with the new traces.

Related to https://github.com/AndreasMadsen/clarify/pull/14

### CODE REVIEW

File: test/test.js [hunk 1]
```diff
@@ -7,16 +7,19 @@ const path = require('path');
...
-  expected: path.resolve(__dirname, 'expected'),
+  expected: EXPECTED_PATH,
   readSource: false,
 
   update: false,
 
   test: function (name, callback) {
     const filepath = path.join(SCRIPTS_PATH, name + '.js');
-    const p = execspawn(`${process.execPath} -r ../trace.js --stack-trace-limit=1000 ${JSON.stringify(filepath)} 2>&1`);
+    const p = execspawn(`${process.execPath} -r ./trace.js --stack-trace-limit=1000 ${JSON.stringify(filepath)} 2>&1`);
```
Comment (afharo) [inline test/test.js:22]: I had to remove the `..` reference to the parent dir to be able to run it via `npm t`

File: test/test.js [hunk 2]
```diff
@@ -5,18 +5,22 @@ const execspawn = require('execspawn');
...
-  expected: path.resolve(__dirname, 'expected'),
+  expected: EXPECTED_PATH,
   readSource: false,
 
   update: false,
 
   test: function (name, callback) {
     const filepath = path.join(SCRIPTS_PATH, name + '.js');
-    const p = execspawn(`${process.execPath} -r ../trace.js --stack-trace-limit=1000 ${JSON.stringify(filepath)} 2>&1`);
+    const p = execspawn(`${process.execPath} -r ${TRACE_PATH} --stack-trace-limit=1000 ${JSON.stringify(filepath)} 2>&1`);
```
Comment (afharo) [inline test/test.js:23]: Running it as `npm t` failed without this change.

**Comment by AndreasMadsen** [comment]
Hi @afharo, I have seen your PR. First of all, thanks :) As this is just a change to the test and I'm very busy this week, I will review it next week. If I somehow forget, please ping me.

**Comment by afharo** [comment]
@AndreasMadsen, totally understandable! Thanks for the quick review and publish on the `clarify` one. I agree that this one is not as critical since it's only changing tests. No rush on my end.

**Comment by AndreasMadsen** [comment]
One thing I'm curious about is all the CallSite objects. They come from `stack-chain` and are thus not directly related to this module. However, it would be nice if the Symbols didn't pollute the output, since it provides no information.

**Comment by afharo** [comment]
> However, it would be nice if the Symbols didn't pollute the output, since it provides no information.

++! Happy to take a look at `stack-chain` to see if we can optimize it for v20.

**Comment by AndreasMadsen** [comment]
Hi @afharo, I went ahead and merged this with some changes.

1. Similar to clarify, internals are now prefixed with `node:internal/`. This module removes the async_hooks bounces, since those are added due to this module. We could maybe do something more clever that detects if there are other hooks, and only remove the CallSites if there are no other hooks. 
2. Node.js v18 is EOL pretty soon. Since it's just a test update, I don't see a need to keep old test files around. I would guess they didn't work with Node.js v18 anyway.
3. It's better to strip the node version from the files, that way it doesn't show up in the git diff next time the test expectations are updated. 

Thanks for the contribution.

**Pull request merged**